pyvista/core/utilities/features.py

```diff
@@ -77,25 +101,61 @@ def voxelize(mesh, density=None, check_surface=True):
     if not surface.is_all_triangles:
         # reduce chance for artifacts, see gh-1743
         surface.triangulate(inplace=True)
-
+    
```

Comment:
Make sure to turn on "Trim trailing whitespace" in your IDE. For example in VSCode:

![hZrUY](https://github.com/pyvista/pyvista/assets/22067021/57f1e4fa-281d-48cd-9927-7a23f7467edd)